lld/test/ELF/linkerscript/discard-section-reloc.s

```diff
@@ -0,0 +1,14 @@
+# REQUIRES: x86
```

Comment:
This is copied from discard-section.s but adds `-r`. We should use the same file: https://maskray.me/blog/2021-08-08-toolchain-testing#i-dont-know-an-existing-test-can-be-enhanced

Reply:
Please remove the new file discard-symbol.s as well.

You can squash the commits and rebase on top of 4fb49f44fdf558a942de2b0fc81e7f1fdf1c798c

Reply:
Rebase on top of 557299c9b6464f27968904aad5429cfb1512434e

`weakref1` is test case about a symbol defined in a discarded section (`/DISCARD/`) that is weak referenced. It's controversial whether we should report an error. Since our implementation reports an error and GNU ld reports an error as well, reporting an error seems fine.